nm/bond: Keep child interface when bond mode change

cathay4t opened the pull request:
Due to NetworkManager bug https://bugzilla.redhat.com/1881318 , the
bond child will be deactivated when bond mode changed.

Workaround is manually activate bond child.

Integration test case included and marked as tier1 as oVirt is depending
on it.

ffmancera reviewed (approved):
Thanks!

EdDev commented:
Are you planning to remove this workaround after it is fixed by NM? If so, how do you track it?

ffmancera commented:
> Are you planning to remove this workaround after it is fixed by NM? If so, how do you track it?

I think we could create an issue and track it there.

tyll commented on libnmstate/nm/profile.py:
```diff
@@ -123,6 +123,7 @@ def _group_profile_by_action_order(self):
     def _prepare_state_for_profiles(self, net_state):
         _preapply_dns_fix_for_profiles(self._ctx, net_state)
         _mark_nm_external_subordinate_changed(self._ctx, net_state)
+        _mark_mode_changed_bond_child_interface_as_changed(net_state)
```
What's wrong with codecov? It is strange that this line is not covered since there is a specific test that should use this code.

ffmancera replied:
Good question, it is on our TO-DO list to investigate what is going on with codecov. It is not working properly. I am going to create an issue

cathay4t commented:
> Are you planning to remove this workaround after it is fixed by NM? If so, how do you track it?

I don't think we have plan to bump our minimum support NM version recently.  So I assume we will need the workaround for a
long time. But I will use https://github.com/nmstate/nmstate/issues/1350 tracking this.

cathay4t committed:
nm/bond: Keep child interface when bond mode change

Due to NetworkManager bug https://bugzilla.redhat.com/1881318 , the
bond child will be deactivated when bond mode changed.

Workaround is manually activate bond child.

Integration test case included and marked as tier1 as oVirt is depending
on it.

Signed-off-by: Gris Ge <fge@redhat.com>
